apps/minifront/src/components/swap/auction-list/claim-all-button.tsx

```diff
@@ -0,0 +1,87 @@
+import { useAuctionInfos } from '../../../state/swap/dutch-auction';
+import { AddressIndex } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
+import { Button } from '@repo/ui/components/ui/button';
+import { AllSlices } from '../../../state';
+import { useStoreShallow } from '../../../utils/use-store-shallow.ts';
+import { filterWithLimit } from './helpers.ts';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@repo/ui/components/ui/tooltip';
+
+const claimAllAuctionButtonSelector = (state: AllSlices) => ({
+  endAllAuctions: state.swap.dutchAuction.endAllAuctions,
+  withdrawAllAuctions: state.swap.dutchAuction.withdrawAllAuctions,
+});
+
+export const ClaimAllAuctionButton = () => {
+  const { endAllAuctions, withdrawAllAuctions } = useStoreShallow(claimAllAuctionButtonSelector);
+  const { data } = useAuctionInfos();
+
+  if (!data?.length) {
+    return null;
+  }
+  if (data.some(a => a.localSeqNum === 0n)) {
+    return (
+      <TooltipProvider>
+        <Tooltip>
+          <TooltipTrigger
+            onClick={() =>
+              void endAllAuctions(
+                // Chain has a transaction size limit, so we can add at most a batch of 48 auctions in a single transaction
+                // see https://github.com/penumbra-zone/web/issues/1166#issuecomment-2263550249
+                filterWithLimit(data, a => a.localSeqNum === 0n, 48),
+                // TODO Should use the index of the selected account after the account selector for the auction is implemented
```

Comment:
Currently, we can end/withdraw all only for Account 0 
see https://github.com/penumbra-zone/web/issues/1166#issuecomment-2265680885

I will open a separate issue for this to add an account selector to the auction list